### PR TITLE
Keycache config

### PR DESCRIPTION
**Pull request by jhiemstrawisc**
First pass at creating keycache config api.

### CODE REVIEW

**Review by bbockelm — CHANGES_REQUESTED** [review]
Second round is looking good!

Only significant request is to change the function-based interface to a generic config key-based interface.

So, instead of:
```
keycache_set_update_interval_s(300, &err)
```

we should have a function call along the lines of:

```
config_set_int("keycache.update_interval_s", 300, &err)
```

This does remove the type safety of the interface but provides more forward/backward compatibility as adding or removing config variable names does not change the API/ABI.

When adding the new config function, please add a unit test covering the case of a non-existing key as well.

Thanks!

**Review by bbockelm — APPROVED** [review]
LGTM.

Fixes #86

**Comment by djw8605** [comment]
There's some minor conflicts from the massive async addition.  Could you resolve the conflicts, then I think we're good to go.

**Comment by djw8605** [comment]
A few build errors.

File: src/scitokens_internal.cpp [hunk 1]
```diff
@@ -666,10 +659,12 @@ Validator::get_public_keys_from_web_continue(std::unique_ptr<AsyncStatus> status
         auto now = std::time(NULL);
         // TODO: take expiration time from the cache-control header in the response.
         
-        int next_update_delta, expiry_delta;
-        get_default_expiry_time(next_update_delta, expiry_delta);
-        status->m_next_update = now + next_update_delta;
-        status->m_expires = now + expiry_delta;
+        int next_update_delta = configurer::Configuration::get_next_update_delta();
+        int expiry_delta = configurer::Configuration::get_expiry_delta();
+        next_update = now + next_update_delta;
```
Comment (djw8605) [inline src/scitokens_internal.cpp:664]: next_update and expires doesn't exist.  Go ahead and put them right into the status->m_next_update and status->m_expires variables.

**Comment by bbockelm** [comment]
@jhiemstrawisc - can you take another look at this one?  Needs a rebase after the big code churns in the last few weeks.

**Comment by jhiemstrawisc** [comment]
Rebase done, checks passed!

**Comment by djw8605** [comment]
I've merged in the linter check, and the code doesn't pass.  It did annotate the pull request, though, without useful comments.

I wonder if auto-fix is the only way forward?  Since annotations do not give suggested changes?

**Comment by jhiemstrawisc** [comment]
I'm closing this PR because the git history was damaged. See the new PR #111 to replace this one.

**Pull request closed without merging**